ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSfilemanip.cpp

```diff
@@ -108,12 +108,11 @@ void ini_section_delete(std::string section)
 /* OS Specific; should be moved */
 
 int file_exists(std::string fname) {
-    DWORD attributes = GetFileAttributes(fname.c_str());
-    if(attributes == INVALID_FILE_ATTRIBUTES) {
-        return 0;
-    } else {
-        return 1;
-    }
+  DWORD file_attr;
```

Comment:
Is this the behavior in GM6? In PHP, for example, `file_exists` returns true for directories. Robert, mind checking that? If so, please move this to `regular_file_exists` or have Sam's code below just check `file_exists() && !directory_exists()`.

Reply:
i find that fucked up and im pretty sure that's how GM works ill test it to be absolutly sure and get back to you but seriously people need a way to distinct the difference between files and directories if they want it to return true for both they can do an "or" if-statement checking the return for both if they really want to

Reply:
@JoshDreamland i was right 

show_message(file_exists("C:\Users\Samuel\Documents\AGameAWeek")); // returns 0
show_message(directory_exists("C:\Users\Samuel\Documents\AGameAWeek")); returns 1

yes the folder existed on my system in both checks

Reply:
No Sam, directories are in fact files, they are even entered into the _file_ table in the _file_ system.

> directory is actually no more than a **file** - Dennis Ritchie (he invented C)1972
https://www.bell-labs.com/usr/dmr/www/notes.html

What Josh said is also true of Java. He did give the correct canonical way of checking if a file is not a directory. I'll be back with GM results.

Reply:
I used the sandboxed working directory in GMSv1.4 for the test. It seems GM is just carrying on this oddity of Delphi and Windows and it's very different from every other programming language in doing so.
```cpp
show_message(string(file_exists("group/")));
show_message(string(directory_exists("group/")));
```
|            | GM5   | GM8   | GMSv1.4 |
|------------|-------|-------|---------|
| File?      | False | False | False   |
| Directory? | True  | True  | True    |